src/Nri/Ui/Table/V8.elm

```diff
@@ -0,0 +1,368 @@
+        columns
+        [ tableHeader columns
+        , tableBody toRow data
+        ]
+
+
+table : List Style -> List (Column data msg) -> List (Html msg) -> Html msg
+table styles columns =
+    Html.table [ css (styles ++ tableStyles columns) ]
+
+
+tableHeader : List (Column data msg) -> Html msg
+tableHeader columns =
+    thead []
+        [ tr [ css headersStyles ]
+            (List.map tableColHeader columns)
+        ]
+
+
+tableColHeader : Column data msg -> Html msg
+tableColHeader (Column header _ width _ sort _ _) =
+    th
+        [ Attributes.scope "col"
+        , css (width :: headerStyles)
+        , Attributes.attribute "aria-sort" <|
+            case sort of
+                Nothing ->
+                    "none"
+
+                Just Ascending ->
+                    "ascending"
+
+                Just Descending ->
+                    "descending"
+        ]
+        [ header ]
+
+
+tableBody : (a -> Html msg) -> List a -> Html msg
+tableBody toRow items =
+    tbody [] (List.map toRow items)
+
+
+
+-- STYLES
+
+
+headersStyles : List Style
+headersStyles =
+    [ -- We use a inset box shadow for a bottom border instead of an actual
+      -- border because with our use of `border-collapse: collapse`, the bottom
+      -- gray border sticks to the table instead of traveling with the header
+      -- when the header has `position: sticky` applied.
+      --
+      -- We add the style on the children instead of on the parent because
+      -- `<tr>` tags do not display box shadows in Safari (although by the time
+      -- you read this, they may have fixed it. Feel free to try again!)
+      Css.Global.children [ Css.Global.th [ boxShadow4 inset (px 0) (px -3) gray75 ] ]
+    , height (px 45)
+    , fontSize (px 15)
+    ]
+
+
+headerStyles : List Style
+headerStyles =
+    [ padding4 (px 11) (px 12) (px 14) (px 12)
+    , textAlign left
+    , fontWeight bold
+    , color gray20
+    ]
+
+
+rowStyles : Bool -> List Style
+rowStyles alternatingRowColors =
+    [ height (px 45)
+    , fontSize (px 14)
+    , color gray20
+    , if alternatingRowColors then
+        pseudoClass "nth-child(odd)"
+            [ backgroundColor gray96 ]
+
+      else
+        Css.batch []
+    ]
+
+
+loadingContentStyles : List Style
+loadingContentStyles =
+    [ width (pct 100)
+    , display inlineBlock
+    , height (Css.em 1)
+    , borderRadius (Css.em 1)
+    , backgroundColor gray75
+    ]
+
+
+loadingCellStyles : List Style
+loadingCellStyles =
+    [ batch flashAnimation
+    , padding2 (px 14) (px 10)
+    ]
+
+
+loadingTableStyles : List Style
+loadingTableStyles =
+    fadeInAnimation
+
+
+tableStyles : List (Column data msg) -> List Style
+tableStyles columns =
+    let
+        hasPlaceholderColumns =
+            List.any (\(Column _ _ _ _ _ _ h) -> h) columns
+    in
+    [ borderCollapse collapse
+    , baseFont
+    , Css.width (Css.pct 100)
+    , if hasPlaceholderColumns then
+        Css.tableLayout Css.fixed
```

Comment:
I think that there's value in supporting `fixed` layout even if we are not using placeholder columns. Should we just make that a separate attribute?

Reply:
sure!

Reply:
Isn't it already supported with additional styles?

Reply:
Oh, you're right. Feel free to ignore this, then.